yt/geometry/particle_geometry_handler.py

```diff
@@ -75,7 +75,9 @@ def _setup_filenames(self):
         cls = self.dataset._file_class
         self.data_files = \
           [cls(self.dataset, self.io, template % {'num':i}, i)
-           for i in range(ndoms)]
+           for i in range(ndoms[0])]
```

Comment:
I think it would be better to say `for i in range(int(ndoms))`, right now `ndoms` is sometimes an integer and sometimes an array scalar and the way you have it here breaks if it's an integer.

I'd also remove the comment below because it's a relatively minor point and I don't think is worth a comment.

Reply:
I see! I did not realise that this was sometimes an integer. I also agree the comment is too much if we use int(). I've made this change below.